src/integration/masternode/application/services/masternode.service.ts

```diff
@@ -61,6 +61,10 @@ export class MasternodeService {
     return this.masternodeRepo.find();
   }
 
+  async getIdleMasternodes(count: number): Promise<Masternode[]> {
+    return this.masternodeRepo.find({ where: { state: MasternodeState.IDLE }, take: count });
```

Comment:
@Kolibri1990 what is the scenario if there is not enough idle masternodes (output count != input count)?

Reply:
Just talked.